app/lib/core/view_models/account_edit_view_model.dart

```diff
@@ -89,4 +93,29 @@ class AccountEditViewModel extends BaseViewModel {
   }
 
   void navigateToAccount() => _navigationService.pop();
+
+  Future<void> removeTransporter(String userId) async =>
+      _databaseService.removeTransporter(userId).catchError((e) => throw (e));
+
+  Future<void> deactivateAccount() async {
+    var dialogResponse = await _dialogService.showConfirmationDialog(
+      title: 'Are you sure?',
+      description: 'Do you really want to deactivate your account??',
+      confirmationText: 'Yes',
+      cancelText: 'No',
+    );
+
+    if (!dialogResponse.confirmed) return;
+    setState(ViewState.Busy);
+    try {
+      _navigationService.navigateTo(WelcomeScreen.route);
+      await _authenticationService.deleteAccount();
```

Comment:
Does the navigation back to welcome screen still occur when deleting the account fails? (You can test this by disconnecting from the internet before hitting the deactivate button)

Reply:
Yes, it goes to welcome screen.

Reply:
Hm, I would prefer it not to back out if failure, but not really important to fix now.